packages/peregrine/src/hooks/useDOMQuery.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect, useCallback } from 'react';
+
+const findDOMElements = querySelector =>
+    document.querySelectorAll(querySelector) || [];
+
+/**
+ * A custom hook that will return the array of elements that
+ * match the querySelector. document.querySelectorAll is used
+ * to find all elements.
+ *
+ * @param {string} querySelector
+ * @return Array<HTMLElement> and {
+ *  setInnerHTML: (string) => void,
+ *  setInnerText: (string) => void,
+ *  setAttribute: (string, any) => void
+ * }
+ */
+export const useDOMQuery = querySelector => {
+    const [elements, setElements] = useState([]);
```

Comment:
Unfortunately, I believe this is a memory leak. Storing references to DOM nodes in state will [prevent them from being collected](https://developers.google.com/web/tools/chrome-devtools/memory-problems/#discover_detached_dom_tree_memory_leaks_with_heap_snapshots), even if React removes those nodes from the DOM.